### PR TITLE
Small improvement to handling of lighting-addresses

### PR DESCRIPTION
**Pull request by nochiel**
This PR depends on https://github.com/breez/breezmobile/pull/563.
- ~Silent fail if lightning-address is not a valid address. This error handling will be improved some more later.~
- [x] Parse the error returned by lnurl.HandleLNUrl and try to give the user a meaningful error.

### CODE REVIEW

File: account/lnurl.go [hunk 1]
```diff
@@ -62,6 +62,25 @@ func (a *Service) HandleLNURL(rawString string) (result *data.LNUrlResponse, err
 
 	rawurl, iparams, err := lnurl.HandleLNURL(encodedLnurl)
 	if err != nil {
+
+		if lightningAddress != "" {
```
Comment (roeierez) [inline account/lnurl.go:66]: It seems weird to have two different types of errors (one for lightning address and one for lnurl).
Why not using the data.LNUrlREsponse with error field set to all cases?

**Review by roeierez — CHANGES_REQUESTED** [review]

File: account/lnurl.go (hunk 1 above)
Comment (nochiel) [reply account/lnurl.go:65]: Agreed. But I've reverted the addition of LNUrlResponseError in the latest commits. I don't think I need it, after all, to handle the errors properly. Hopefully the new method is better.

File: account/lnurl.go [hunk 2]
```diff
@@ -62,7 +62,61 @@ func (a *Service) HandleLNURL(rawString string) (result *data.LNUrlResponse, err
 
 	rawurl, iparams, err := lnurl.HandleLNURL(encodedLnurl)
 	if err != nil {
-		return nil, fmt.Errorf("%v : %w", handleLNUrlError, err)
+		if lightningAddress != "" {
+
+			const unknownResponseTagError = "unknown response tag "
+			if strings.HasPrefix(err.Error(), unknownResponseTagError) {
```
Comment (roeierez) [inline account/lnurl.go:68]: Is all this only to show a more accurate error? I am not sure it worth it and definitely doesn't justify an extra http request.

**Review by roeierez — CHANGES_REQUESTED** [review]

File: account/lnurl.go (hunk 2 above)
Comment (nochiel) [reply account/lnurl.go:68]: Yes, that's the only reason. I'm happy to delete it though.

Comment (kingonly) [reply account/lnurl.go:68]: Let's simply as Roei suggests.

**Review by roeierez — APPROVED** [review]

**Pull request merged**